File: _legacy/controller/categories/delete_item.php
<?php
namespace Core;
use Exception, Controller\Main;
use Controller\categories\UtilForCategories;

if (!defined('__API_GOOSE__')) exit();

/**
 * delete category
 *
 * @var Goose|Connect $this
 */

try
{
  // check and set srl
  if (($srl = (int)($this->params['srl'] ?? 0)) <= 0)
  {
    throw new Exception(Message::make('error.notFound', 'srl'));
  }

  // connect db
  $this->model->connect();

  // check access
  $token = Main::checkAccessItem($this, (object)[
    'table' => 'categories',
    'srl' => $srl,
  ]);

  // get item
  $category = $this->model->getItem((object)[
    'table' => 'categories',
    'where' => 'srl='.$srl,
  ])->data;

  // update article items
  $tableName = match ($category->module)
  {
    UtilForCategories::$module['article'] => 'articles',
    UtilForCategories::$module['json'] => 'json',
  };
  if ($tableName ?? false)
  {
    $this->model->edit((object)[
      'table' => $tableName,
      'data' => [ 'category_srl=NULL' ],
      'where' => 'category_srl='.$srl,
      'continue' => true,
      'debug' => __API_DEBUG__,
    ]);
  }

  // remove data
  $output = Main::delete($this, (object)[
    'table' => 'categories',
    'srl' => $srl,
  ]);

  // set output
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  return Output::result($output);
}
catch (Exception $e)
{
  if (isset($this->model)) $this->model->disconnect();
  return Error::result($e->getMessage(), $e->getCode());
}

File: src/endpoints/category/delete_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token

async def delete_item(params: dict = {}, req = None, _db: DB = None):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        checking_token(req, db)

        # set where
        where = [ f'srl = {params.srl}' ]

        # check item
        count = db.get_count(
            table_name = Table.CATEGORY.value,
            where = where,
        )
        if count == 0: raise Exception('Item not found.', 204)

        # delete item
        db.delete_item(
            table_name = Table.CATEGORY.value,
            where = where,
        )

        # set result
        result = output.success({
            'message': 'Complete delete category.',
        })
    except Exception as e:
        result = output.exc(e)
    finally:
        if not _db and db: db.disconnect()
        return result
